File: FigurasGeometricas/EntidadesVb/Circulo.vb
﻿Public Class Circulo
	Inherits Figura
	Implements Figura2D

	Public Property Radio As Single
	Public Sub New(nombre As String, radio As Single)
		MyBase.New(nombre)
		Me.Radio = radio
	End Sub
	Public Overrides Function ToString() As String
		Return $"Soy {Nombre} y tengo el radio de {Radio}"
	End Function

	Public Function CalcularArea() As Single Implements Figura2D.CalcularArea
		Return Math.PI * Math.Pow(Radio, 2)
	End Function
End Class

File: FigurasGeometricas/EntidadesCs/Circulo.cs
namespace EntidadesCs
{
	public class Circulo : Figura, Figura2D
	{
		public float Radio {  get; set; }
		public Circulo(string nombre, float radio) : base(nombre)
		{
			Radio = radio;
		}
		public float CalcularArea()
		{
			return (float)(Math.PI * Math.Pow(Radio, 2));
		}
		public override string ToString()
		{
			return $"Soy {Nombre} y tengo mi radio de {Radio}";
		}
	}
}
